src/__tests__/components/fields/image-upload/image-upload.spec.tsx

```diff
@@ -0,0 +1,884 @@
+	eventType: string;
+	eventListener: (this: Element, ev: Event) => any;
+}
+const FrontendEngineWithEventListener = (props: ICustomFrontendEngineProps) => {
+	const { eventType, eventListener, ...otherProps } = props;
+	const formRef = useRef<IFrontendEngineRef>();
+	useEffect(() => {
+		if (eventType && eventListener) {
+			const currentFormRef = formRef.current;
+			currentFormRef.addFieldEventListener(eventType, "field", eventListener);
+			return () => currentFormRef.removeFieldEventListener(eventType, "field", eventListener);
+		}
+	}, [eventListener, eventType]);
+
+	return <FrontendEngine {...otherProps} ref={formRef} />;
+};
+
+interface IRenderAndPerformActionsOptions {
+	overrideField?: TOverrideField<IImageUploadSchema>;
+	overrideSchema?: TOverrideSchema;
+	files?: { name: string; type: string }[];
+	uploadType?: "input" | "drag & drop";
+	reviewImage?: boolean;
+	eventType?: string;
+	eventListener?: (this: Element, ev: Event) => any;
+}
+
+/**
+ * render component
+ * upload with file(s) specified
+ * optionally go to review modal
+ */
+const renderComponent = async (options: IRenderAndPerformActionsOptions = {}) => {
+	jest.spyOn(ImageHelper, "convertBlob").mockResolvedValue(JPG_BASE64);
+	jest.spyOn(FileHelper, "getMimeType").mockResolvedValue("image/jpeg");
+
+	const {
+		overrideField,
+		overrideSchema,
+		eventType,
+		eventListener,
+		files = [],
+		uploadType = "input",
+		reviewImage,
+	} = options;
+	const json: IFrontendEngineData = {
+		id: FRONTEND_ENGINE_ID,
+		sections: {
+			section: {
+				uiType: "section",
+				children: {
+					[COMPONENT_ID]: {
+						label: "Image Upload",
+						uiType: UI_TYPE,
+						uploadOnAddingFile: {
+							method: "post",
+							url: "test",
+						},
+						...overrideField,
+					},
+					...getSubmitButtonProps(),
+					...getResetButtonProps(),
+				},
+			},
+		},
+		...overrideSchema,
+	};
+	render(
+		<FrontendEngineWithEventListener
+			data={json}
+			onSubmit={SUBMIT_FN}
+			eventType={eventType}
+			eventListener={eventListener}
+		/>
+	);
+
+	const uploadField = await waitFor(() =>
+		uploadType === "input" ? getDragInputUploadField() : screen.getByTestId("field-drag-upload")
+	);
+	await act(async () => {
+		for (let i = 0; i < files.length; i++) {
+			fireEvent[uploadType === "input" ? "change" : "drop"](uploadField, {
+				target: {
+					files: [files[i]],
+				},
+			});
+			await flushPromise();
+		}
+	});
+
+	if (reviewImage) {
+		await waitFor(() => fireEvent.click(getField("button", "Ok")));
+		await new Promise((resolve) => setTimeout(resolve));
+	}
+};
+
+describe("image-upload", () => {
+	beforeEach(() => {
+		jest.clearAllMocks();
+		jest.restoreAllMocks();
+		setupJestCanvasMock();
+
+		jest.spyOn(FileHelper, "truncateFileName").mockImplementation((fileName) => fileName);
+		uploadSpy = jest.spyOn(AxiosApiClient.prototype, "post").mockResolvedValue({ id: 1 });
+	});
+
+	it("should be able to render the field", async () => {
```

Comment:
this test is not passing for me
`Unable to find an accessible element with the role "button" and name "Add photos"`

Reply:
I guess it's because the subtitle label overrode the button's implicit name. will check on the change when @weili-govtech is back

Reply:
seems like the assertion is failing because I added a label element for the button, somehow react testing-library stopped recognising it as a button. in view of this, I have to `getByLabelText` to find that button instead. this will be part of a new PR